src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -94,17 +96,26 @@ public class MeterProviderSdk
                 {
                     if (meterSourcesToSubscribe.ContainsKey(instrument.Meter.Name))
                     {
-                        var index = Interlocked.Increment(ref this.metricIndex);
-                        if (index >= MaxMetrics)
+                        // lookup views for this instrument (and meter,type etc).
```

Comment:
if no views, follow existing approach. Or we can insert a view with defaults, so if NO view, drop the instrument